### DISCUSSION TITLE
Would anyone need a _click keyword to automate mouse clicks based on a condition?

### DISCUSSION THREAD

**Discussion by cortexrd** [cortexrd/Knack-Toolkit-Library#174] [Ideas]
I am brainstorming on some automation ideas, to enable mouse clicks on action links for example.

Use cases:

1. One of my first use would be to handle mailing lists by sending emails in sequence from a grid until all are sent.
2. Soft delete, archive and hard delete of records
3. Navigating to another page
4. Forcing a logout

Of course for those actions that must be executed only once, we'd need a flag of some kind, like a Date not blank, Sent (yes/no) or other.

Do you have other use cases for this?

Norm

**Comment by CSWinnall** [comment]
I can certainly see some uses for this.  Especially with archiving records.  I think with a hard delete I'd be worried about leaving this to automation unless there were some strict controls i.e. a confirmation which I think would negate the need for an automation.

Navigation to another page is interesting I think we may find that useful.
